hbase-server/src/main/java/org/apache/hadoop/hbase/quotas/DefaultOperationQuota.java

```diff
@@ -122,6 +126,9 @@ public void close() {
 
   @Override
   public long getReadAvailable() {
+    if (readAvailable == Long.MAX_VALUE && readCapacityUnitAvailable != Long.MAX_VALUE) {
```

Comment:
Mind explaing a bit more here? It will also effect the quota for operations other than scan?

Reply:
Scan use this value to limit the result size:
`   long maxQuotaResultSize = Math.min(maxScannerResultSize, quota.getReadAvailable());
    long maxResultSize;
    if (scanner.getMaxResultSize() > 0) {
      maxResultSize = Math.min(scanner.getMaxResultSize(), maxQuotaResultSize);
    } else {
      maxResultSize = maxQuotaResultSize;
    }`

Now it only considers the 'read size' type quota, and if it is not set, the scan can not be limited. The 'read capacity unit' is also a type of size quota, we can use it to calculate the available read size.

Reply:
Now only scan use it.